app/controllers/api/v1/application_controller.rb

```diff
@@ -2,12 +2,27 @@ module API
   module V1
     class ApplicationController < ::ApplicationController
       before_action -> { skip_authorization }
+      before_action :check_recruitment_cycle_is_current_year
 
       def authenticate
         authenticate_or_request_with_http_token do |token|
           ActiveSupport::SecurityUtils.secure_compare(token, Rails.application.config.authentication_token)
         end
       end
+
+      def check_recruitment_cycle_is_current_year
+        current_year = Settings.current_recruitment_cycle_year
+        next_year = current_year + 1
+
+        requested_year = params[:recruitment_year]
+        is_next_or_current_year = [current_year, next_year]
```

Comment:
``` ruby
RecruitmentCycle.current_recruitment_cycle
    RecruitmentCycle.next_recruitment_cycle
```

don't you wanna use one of these?

Reply:
Yes

Reply:
Wow! That's a rabbit hole. It breaks a bunch of different tests because `RecruitmentCycle.next_recruitment_cycle` is nil. Not sure if we should do this right now.